### PR TITLE
Fine-grained and ordered ECMP types

### PR DESCRIPTION
**Pull request by marian-pritsak**

### CODE REVIEW

File: inc/sainexthopgroup.h [hunk 1]
```diff
@@ -219,6 +255,33 @@ typedef enum _sai_next_hop_group_member_attr_t
...
+     * @default 0
+     */
+    SAI_NEXT_HOP_GROUP_MEMBER_ATTR_INDEX,
+
+    /**
+     * @brief Object priority for enforcing the members' order.
+     *
+     * Index specifying the member's order. The index is not strict exhibiting the behavior of
+     * the priority. Members don't have to have the sequential priorities and it's driver's job
+     * to translate the priorities to the real indeces in the group.
```
Comment (itaibaz) [inline inc/sainexthopgroup.h:276]: The plural is indices and not indeces, and then perhaps the word won't be needed to be added to the dictionary

File: inc/sainexthopgroup.h [hunk 2]
```diff
@@ -133,6 +142,33 @@ typedef enum _sai_next_hop_group_attr_t
      */
     SAI_NEXT_HOP_GROUP_ATTR_COUNTER_ID,
 
+    /**
+     * @brief Configured group size
+     *
+     * Maximum desired number of members. The real size should
+     * be queried from SAI_NEXT_HOP_GROUP_ATTR_ACTUAL_SIZE
```
Comment (itaibaz) [inline inc/sainexthopgroup.h:149]: Did you mean SAI_NEXT_HOP_GROUP_ATTR_REAL_SIZE and not SAI_NEXT_HOP_GROUP_ATTR_ACTUAL_SIZE ?

File: inc/sainexthopgroup.h [hunk 3]
```diff
@@ -133,6 +142,33 @@ typedef enum _sai_next_hop_group_attr_t
...
+     *
+     * @type sai_uint32_t
+     * @flags CREATE_ONLY
+     * @validonly SAI_NEXT_HOP_GROUP_ATTR_TYPE == SAI_NEXT_HOP_GROUP_TYPE_FINE_GRAIN_ECMP
+     * @isresourcetype true
```
Comment (itaibaz) [inline inc/sainexthopgroup.h:154]: Can you explain the intention here for marking as resource -
Is it querying a specific next hop group object to see how many entries are still available ? 
or is the intention to query how many available next hop groups of a certain configured size (and optionally also type) can still be created ?

File: inc/sainexthopgroup.h (hunk 2 above)
Comment (marian-pritsak) [reply inc/sainexthopgroup.h]: Yes, thanks

File: inc/sainexthopgroup.h (hunk 1 above)
Comment (marian-pritsak) [reply inc/sainexthopgroup.h]: Done

File: inc/sainexthopgroup.h (hunk 3 above)
Comment (marian-pritsak) [reply inc/sainexthopgroup.h:155]: To query how many NH groups of a certain size can be crated.

**Pull request merged**